### ISSUE TITLE
[BUG] Mojo Language Server crashes in Visual Studio Code

### ISSUE THREAD

**Issue by ivellapillil** [modularml/mojo#2446]
### Bug description

Editing Mojo source code in Visual Studio Code results in crash of Mojo Language Server.

### Steps to reproduce

Try editing the following code in VSCode

```mojo
 
fn exec_rt_closure(x: Int, bin_op_cl: fn(Int) escaping -> None): pass
 
fn main():
    var i: Int = 10
    fn ander(x: Int): # the input argument is needed for the crash
        print(i) # Accessing i results in the crash.
    exec_rt_closure(10, ander)
 
```

MLS crashes with the following error:

```
["INFO" - 20:09:34] Language Server started
I[20:09:34.582] --> initialize(0)
I[20:09:34.583] --> reply:(0)
D[20:09:34.583] >>> {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix","refactor","info"]},"completionProvider":{"allCommitCharacters":["\t","."],"resolveProvider":false,"triggerCharacters":["."]},"definitionProvider":true,"documentSymbolProvider":true,"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"notebookDocumentSync":{"notebookSelector":[{"cells":[{"language":"mojo"}],"notebook":{"notebookType":"jupyter-notebook","scheme":"file"}}]},"referencesProvider":true,"semanticTokensProvider":{"full":{"delta":true},"legend":{"tokenModifiers":[],"tokenTypes":["variable","specialVariable","parameter","function","method","property","class","interface","type","namespace"]},"range":false},"signatureHelpProvider":{"triggerCharacters":["(","[",","]},"textDocumentSync":{"change":2,"openClose":true,"save":true}},"serverInfo":{"name":"mojo-lsp-server","version":"0.0.1"}}}
 
I[20:09:34.612] --> initialized
I[20:09:34.629] --> textDocument/didOpen
D[20:09:34.629] --> textDocument/didOpen: uri='try_out.mojo', version=1
I[20:09:34.632] --> textDocument/inlayHint(1)
I[20:09:34.694] --> textDocument/publishDiagnostics
D[20:09:34.694] >>> {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"try_out.mojo","version":1}}
 
I[20:09:34.694] --> reply:(1)
D[20:09:34.694] >>> {"id":1,"jsonrpc":"2.0","result":[]}
 
I[20:09:34.763] --> textDocument/codeAction(2)
I[20:09:34.764] --> reply:(2)
D[20:09:34.764] >>> {"id":2,"jsonrpc":"2.0","result":[]}
 
I[20:09:34.947] --> textDocument/inlayHint(3)
I[20:09:34.947] --> reply:(3)
D[20:09:34.947] >>> {"id":3,"jsonrpc":"2.0","result":[]}
 
I[20:09:34.964] --> textDocument/documentSymbol(4)
I[20:09:34.964] --> textDocument/foldingRange(5)
I[20:09:34.964] --> reply:(5)
D[20:09:34.964] >>> {"id":5,"jsonrpc":"2.0","result":[]}
 
I[20:09:34.964] --> reply:(4)
D[20:09:34.964] >>> {"id":4,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:35.049] --> textDocument/codeAction(6)
I[20:09:35.049] --> reply:(6)
D[20:09:35.049] >>> {"id":6,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.080] --> textDocument/semanticTokens/full(7)
Assertion failed: (isa(Val) && "cast() argument of incompatible type!"), function cast, file Casting.h, line 572.
Please submit a bug report to https://github.com/modularml/mojo/issues and include the crash backtrace along with all the relevant source codes with the contents they had at crash time.
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0 mojo-lsp-server 0x0000000102262d4c llvm_strlcpy + 44160
1 mojo-lsp-server 0x0000000102260eac llvm_strlcpy + 36320
2 mojo-lsp-server 0x00000001022633ec llvm_strlcpy + 45856
3 libsystem_platform.dylib 0x000000018bc2b584 _sigtramp + 56
4 libsystem_pthread.dylib 0x000000018bbfac20 pthread_kill + 288
5 libsystem_c.dylib 0x000000018bb07a20 abort + 180
6 libsystem_c.dylib 0x000000018bb06d10 err + 0
7 mojo-lsp-server 0x0000000102ebeba0 mbedtls_ssl_handshake_server_step + 2732628
8 mojo-lsp-server 0x00000001023104d4 llvm_strlcpy + 754696
9 mojo-lsp-server 0x00000001021c1624
10 mojo-lsp-server 0x00000001021db324
11 mojo-lsp-server 0x0000000102c5f674 mbedtls_ssl_handshake_server_step + 245032
12 mojo-lsp-server 0x0000000102c5f390 mbedtls_ssl_handshake_server_step + 244292
13 mojo-lsp-server 0x0000000102c5f42c mbedtls_ssl_handshake_server_step + 244448
14 libsystem_pthread.dylib 0x000000018bbfaf94 _pthread_start + 136
15 libsystem_pthread.dylib 0x000000018bbf5d34 thread_start + 8
I[20:09:35.091] --> $/setTrace
[Info - 20:09:35] Connection to server got closed. Server will restart.
I[20:09:35.140] --> initialize(0)
I[20:09:35.140] --> reply:(0)
D[20:09:35.140] >>> {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix","refactor","info"]},"completionProvider":{"allCommitCharacters":["\t","."],"resolveProvider":false,"triggerCharacters":["."]},"definitionProvider":true,"documentSymbolProvider":true,"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"notebookDocumentSync":{"notebookSelector":[{"cells":[{"language":"mojo"}],"notebook":{"notebookType":"jupyter-notebook","scheme":"file"}}]},"referencesProvider":true,"semanticTokensProvider":{"full":{"delta":true},"legend":{"tokenModifiers":[],"tokenTypes":["variable","specialVariable","parameter","function","method","property","class","interface","type","namespace"]},"range":false},"signatureHelpProvider":{"triggerCharacters":["(","[",","]},"textDocumentSync":{"change":2,"openClose":true,"save":true}},"serverInfo":{"name":"mojo-lsp-server","version":"0.0.1"}}}
 
I[20:09:35.140] --> initialized
I[20:09:35.141] --> textDocument/didOpen
D[20:09:35.141] --> textDocument/didOpen: uri='try_out.mojo', version=1
I[20:09:35.153] --> textDocument/codeAction(1)
I[20:09:35.157] --> textDocument/inlayHint(2)
I[20:09:35.181] --> $/cancelRequest
I[20:09:35.182] --> textDocument/codeAction(3)
I[20:09:35.182] --> $/cancelRequest
I[20:09:35.182] --> textDocument/codeAction(4)
I[20:09:35.202] --> textDocument/publishDiagnostics
D[20:09:35.202] >>> {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"try_out.mojo","version":1}}
 
I[20:09:35.202] --> reply:(1)
D[20:09:35.202] >>> {"id":1,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.202] --> reply:(4)
D[20:09:35.202] >>> {"id":4,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.202] --> reply:(2)
D[20:09:35.203] >>> {"id":2,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.203] --> reply:(3)
D[20:09:35.203] >>> {"id":3,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.213] --> textDocument/documentSymbol(5)
I[20:09:35.213] --> reply:(5)
D[20:09:35.213] >>> {"id":5,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:35.245] --> $/setTrace
I[20:09:35.453] --> textDocument/documentSymbol(6)
I[20:09:35.453] --> textDocument/foldingRange(7)
I[20:09:35.453] --> reply:(7)
D[20:09:35.453] >>> {"id":7,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.453] --> reply:(6)
D[20:09:35.454] >>> {"id":6,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:35.604] --> textDocument/semanticTokens/full(8)
Assertion failed: (isa(Val) && "cast() argument of incompatible type!"), function cast, file Casting.h, line 572.
Please submit a bug report to https://github.com/modularml/mojo/issues and include the crash backtrace along with all the relevant source codes with the contents they had at crash time.
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0 mojo-lsp-server 0x0000000104c16d4c llvm_strlcpy + 44160
1 mojo-lsp-server 0x0000000104c14eac llvm_strlcpy + 36320
2 mojo-lsp-server 0x0000000104c173ec llvm_strlcpy + 45856
3 libsystem_platform.dylib 0x000000018bc2b584 _sigtramp + 56
4 libsystem_pthread.dylib 0x000000018bbfac20 pthread_kill + 288
5 libsystem_c.dylib 0x000000018bb07a20 abort + 180
6 libsystem_c.dylib 0x000000018bb06d10 err + 0
7 mojo-lsp-server 0x0000000105872ba0 mbedtls_ssl_handshake_server_step + 2732628
8 mojo-lsp-server 0x0000000104cc44d4 llvm_strlcpy + 754696
9 mojo-lsp-server 0x0000000104b75624
10 mojo-lsp-server 0x0000000104b8f324
11 mojo-lsp-server 0x0000000105613674 mbedtls_ssl_handshake_server_step + 245032
12 mojo-lsp-server 0x0000000105613390 mbedtls_ssl_handshake_server_step + 244292
13 mojo-lsp-server 0x000000010561342c mbedtls_ssl_handshake_server_step + 244448
14 libsystem_pthread.dylib 0x000000018bbfaf94 _pthread_start + 136
15 libsystem_pthread.dylib 0x000000018bbf5d34 thread_start + 8
[Info - 20:09:35] Connection to server got closed. Server will restart.
I[20:09:35.661] --> initialize(0)
I[20:09:35.661] --> reply:(0)
D[20:09:35.661] >>> {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix","refactor","info"]},"completionProvider":{"allCommitCharacters":["\t","."],"resolveProvider":false,"triggerCharacters":["."]},"definitionProvider":true,"documentSymbolProvider":true,"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"notebookDocumentSync":{"notebookSelector":[{"cells":[{"language":"mojo"}],"notebook":{"notebookType":"jupyter-notebook","scheme":"file"}}]},"referencesProvider":true,"semanticTokensProvider":{"full":{"delta":true},"legend":{"tokenModifiers":[],"tokenTypes":["variable","specialVariable","parameter","function","method","property","class","interface","type","namespace"]},"range":false},"signatureHelpProvider":{"triggerCharacters":["(","[",","]},"textDocumentSync":{"change":2,"openClose":true,"save":true}},"serverInfo":{"name":"mojo-lsp-server","version":"0.0.1"}}}
 
I[20:09:35.662] --> initialized
I[20:09:35.663] --> textDocument/didOpen
D[20:09:35.663] --> textDocument/didOpen: uri='try_out.mojo', version=1
I[20:09:35.664] --> textDocument/codeAction(1)
I[20:09:35.669] --> textDocument/inlayHint(2)
I[20:09:35.719] --> textDocument/publishDiagnostics
D[20:09:35.719] >>> {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"try_out.mojo","version":1}}
 
I[20:09:35.719] --> reply:(1)
D[20:09:35.719] >>> {"id":1,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.719] --> reply:(2)
D[20:09:35.719] >>> {"id":2,"jsonrpc":"2.0","result":[]}
 
I[20:09:35.965] --> textDocument/documentSymbol(3)
I[20:09:35.965] --> reply:(3)
D[20:09:35.965] >>> {"id":3,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:35.965] --> textDocument/foldingRange(4)
I[20:09:35.965] --> reply:(4)
D[20:09:35.965] >>> {"id":4,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.115] --> textDocument/semanticTokens/full(5)
Assertion failed: (isa(Val) && "cast() argument of incompatible type!"), function cast, file Casting.h, line 572.
Please submit a bug report to https://github.com/modularml/mojo/issues and include the crash backtrace along with all the relevant source codes with the contents they had at crash time.
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0 mojo-lsp-server 0x000000010304ed4c llvm_strlcpy + 44160
1 mojo-lsp-server 0x000000010304ceac llvm_strlcpy + 36320
2 mojo-lsp-server 0x000000010304f3ec llvm_strlcpy + 45856
3 libsystem_platform.dylib 0x000000018bc2b584 _sigtramp + 56
4 libsystem_pthread.dylib 0x000000018bbfac20 pthread_kill + 288
5 libsystem_c.dylib 0x000000018bb07a20 abort + 180
6 libsystem_c.dylib 0x000000018bb06d10 err + 0
7 mojo-lsp-server 0x0000000103caaba0 mbedtls_ssl_handshake_server_step + 2732628
8 mojo-lsp-server 0x00000001030fc4d4 llvm_strlcpy + 754696
9 mojo-lsp-server 0x0000000102fad624
10 mojo-lsp-server 0x0000000102fc7324
11 mojo-lsp-server 0x0000000103a4b674 mbedtls_ssl_handshake_server_step + 245032
12 mojo-lsp-server 0x0000000103a4b390 mbedtls_ssl_handshake_server_step + 244292
13 mojo-lsp-server 0x0000000103a4b42c mbedtls_ssl_handshake_server_step + 244448
14 libsystem_pthread.dylib 0x000000018bbfaf94 _pthread_start + 136
15 libsystem_pthread.dylib 0x000000018bbf5d34 thread_start + 8
[Info - 20:09:36] Connection to server got closed. Server will restart.
I[20:09:36.168] --> initialize(0)
I[20:09:36.168] --> reply:(0)
D[20:09:36.168] >>> {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix","refactor","info"]},"completionProvider":{"allCommitCharacters":["\t","."],"resolveProvider":false,"triggerCharacters":["."]},"definitionProvider":true,"documentSymbolProvider":true,"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"notebookDocumentSync":{"notebookSelector":[{"cells":[{"language":"mojo"}],"notebook":{"notebookType":"jupyter-notebook","scheme":"file"}}]},"referencesProvider":true,"semanticTokensProvider":{"full":{"delta":true},"legend":{"tokenModifiers":[],"tokenTypes":["variable","specialVariable","parameter","function","method","property","class","interface","type","namespace"]},"range":false},"signatureHelpProvider":{"triggerCharacters":["(","[",","]},"textDocumentSync":{"change":2,"openClose":true,"save":true}},"serverInfo":{"name":"mojo-lsp-server","version":"0.0.1"}}}
 
I[20:09:36.169] --> initialized
I[20:09:36.170] --> textDocument/didOpen
D[20:09:36.170] --> textDocument/didOpen: uri='try_out.mojo', version=1
I[20:09:36.171] --> textDocument/codeAction(1)
I[20:09:36.172] --> textDocument/inlayHint(2)
I[20:09:36.224] --> textDocument/publishDiagnostics
D[20:09:36.225] >>> {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"try_out.mojo","version":1}}
 
I[20:09:36.225] --> reply:(1)
D[20:09:36.225] >>> {"id":1,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.225] --> reply:(2)
D[20:09:36.225] >>> {"id":2,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.474] --> textDocument/documentSymbol(3)
I[20:09:36.475] --> reply:(3)
I[20:09:36.475] --> textDocument/foldingRange(4)
D[20:09:36.475] >>> {"id":3,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:36.475] --> reply:(4)
D[20:09:36.475] >>> {"id":4,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.534] --> textDocument/codeAction(5)
I[20:09:36.534] --> reply:(5)
D[20:09:36.534] >>> {"id":5,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.534] --> textDocument/documentSymbol(6)
I[20:09:36.534] --> reply:(6)
D[20:09:36.534] >>> {"id":6,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:36.591] --> $/setTrace
I[20:09:36.665] --> textDocument/semanticTokens/full(7)
Assertion failed: (isa(Val) && "cast() argument of incompatible type!"), function cast, file Casting.h, line 572.
Please submit a bug report to https://github.com/modularml/mojo/issues and include the crash backtrace along with all the relevant source codes with the contents they had at crash time.
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0 mojo-lsp-server 0x00000001047b2d4c llvm_strlcpy + 44160
1 mojo-lsp-server 0x00000001047b0eac llvm_strlcpy + 36320
2 mojo-lsp-server 0x00000001047b33ec llvm_strlcpy + 45856
3 libsystem_platform.dylib 0x000000018bc2b584 _sigtramp + 56
4 libsystem_pthread.dylib 0x000000018bbfac20 pthread_kill + 288
5 libsystem_c.dylib 0x000000018bb07a20 abort + 180
6 libsystem_c.dylib 0x000000018bb06d10 err + 0
7 mojo-lsp-server 0x000000010540eba0 mbedtls_ssl_handshake_server_step + 2732628
8 mojo-lsp-server 0x00000001048604d4 llvm_strlcpy + 754696
9 mojo-lsp-server 0x0000000104711624
10 mojo-lsp-server 0x000000010472b324
11 mojo-lsp-server 0x00000001051af674 mbedtls_ssl_handshake_server_step + 245032
12 mojo-lsp-server 0x00000001051af390 mbedtls_ssl_handshake_server_step + 244292
13 mojo-lsp-server 0x00000001051af42c mbedtls_ssl_handshake_server_step + 244448
14 libsystem_pthread.dylib 0x000000018bbfaf94 _pthread_start + 136
15 libsystem_pthread.dylib 0x000000018bbf5d34 thread_start + 8
[Info - 20:09:36] Connection to server got closed. Server will restart.
I[20:09:36.721] --> initialize(0)
I[20:09:36.721] --> reply:(0)
D[20:09:36.721] >>> {"id":0,"jsonrpc":"2.0","result":{"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix","refactor","info"]},"completionProvider":{"allCommitCharacters":["\t","."],"resolveProvider":false,"triggerCharacters":["."]},"definitionProvider":true,"documentSymbolProvider":true,"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"notebookDocumentSync":{"notebookSelector":[{"cells":[{"language":"mojo"}],"notebook":{"notebookType":"jupyter-notebook","scheme":"file"}}]},"referencesProvider":true,"semanticTokensProvider":{"full":{"delta":true},"legend":{"tokenModifiers":[],"tokenTypes":["variable","specialVariable","parameter","function","method","property","class","interface","type","namespace"]},"range":false},"signatureHelpProvider":{"triggerCharacters":["(","[",","]},"textDocumentSync":{"change":2,"openClose":true,"save":true}},"serverInfo":{"name":"mojo-lsp-server","version":"0.0.1"}}}
 
I[20:09:36.722] --> initialized
I[20:09:36.723] --> textDocument/didOpen
D[20:09:36.723] --> textDocument/didOpen: uri='try_out.mojo', version=1
I[20:09:36.724] --> textDocument/documentSymbol(1)
I[20:09:36.724] --> textDocument/codeAction(2)
I[20:09:36.725] --> textDocument/inlayHint(3)
I[20:09:36.777] --> textDocument/publishDiagnostics
D[20:09:36.777] >>> {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"try_out.mojo","version":1}}
 
I[20:09:36.777] --> reply:(2)
D[20:09:36.777] >>> {"id":2,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.777] --> reply:(3)
D[20:09:36.777] >>> {"id":3,"jsonrpc":"2.0","result":[]}
 
I[20:09:36.777] --> reply:(1)
D[20:09:36.777] >>> {"id":1,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:37.025] --> textDocument/documentSymbol(4)
I[20:09:37.025] --> reply:(4)
D[20:09:37.025] >>> {"id":4,"jsonrpc":"2.0","result":[{"detail":"exec_rt_closure(x: Int, bin_op_cl: fn(Int, /) escaping -> None)","kind":12,"name":"exec_rt_closure","range":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}},"selectionRange":{"end":{"character":18,"line":0},"start":{"character":3,"line":0}}},{"detail":"main()","kind":12,"name":"main","range":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}},"selectionRange":{"end":{"character":7,"line":2},"start":{"character":3,"line":2}}}]}
 
I[20:09:37.025] --> textDocument/foldingRange(5)
I[20:09:37.025] --> reply:(5)
D[20:09:37.025] >>> {"id":5,"jsonrpc":"2.0","result":[]}
 
I[20:09:37.178] --> textDocument/semanticTokens/full(6)
Assertion failed: (isa(Val) && "cast() argument of incompatible type!"), function cast, file Casting.h, line 572.
Please submit a bug report to https://github.com/modularml/mojo/issues and include the crash backtrace along with all the relevant source codes with the contents they had at crash time.
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0 mojo-lsp-server 0x000000010023ed4c llvm_strlcpy + 44160
1 mojo-lsp-server 0x000000010023ceac llvm_strlcpy + 36320
2 mojo-lsp-server 0x000000010023f3ec llvm_strlcpy + 45856
3 libsystem_platform.dylib 0x000000018bc2b584 _sigtramp + 56
4 libsystem_pthread.dylib 0x000000018bbfac20 pthread_kill + 288
5 libsystem_c.dylib 0x000000018bb07a20 abort + 180
6 libsystem_c.dylib 0x000000018bb06d10 err + 0
7 mojo-lsp-server 0x0000000100e9aba0 mbedtls_ssl_handshake_server_step + 2732628
8 mojo-lsp-server 0x00000001002ec4d4 llvm_strlcpy + 754696
9 mojo-lsp-server 0x000000010019d624
10 mojo-lsp-server 0x00000001001b7324
11 mojo-lsp-server 0x0000000100c3b674 mbedtls_ssl_handshake_server_step + 245032
12 mojo-lsp-server 0x0000000100c3b390 mbedtls_ssl_handshake_server_step + 244292
13 mojo-lsp-server 0x0000000100c3b42c mbedtls_ssl_handshake_server_step + 244448
14 libsystem_pthread.dylib 0x000000018bbfaf94 _pthread_start + 136
15 libsystem_pthread.dylib 0x000000018bbf5d34 thread_start + 8
[Error - 20:09:37] The Mojo Language Client server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
 
```

### System information

```shell
System Version: macOS 14.1 (23B74)
Kernel Version: Darwin 23.1.0
Model Name: MacBook Pro
Model Identifier: MacBookPro18,3
Model Number: MKGP3SM/A
Chip: Apple M1 Pro
Total Number of Cores: 8 (6 performance and 2 efficiency)
Memory: 16 GB

mojo 24.2.1 (2f0dcf11)
modular 0.6.0 (04c05243)
VSCode Plugin: v24.2.1
```

**Comment by ematejska** [comment]
Thanks for filing this issue!

**Comment by River707** [comment]
Thanks for filing, this should be fixed in the upcoming 24.3 release!
